### ISSUE TITLE
automatically rewrite row templates that are still the default generated versions

### ISSUE THREAD

**Issue by mhl** [everypolitician/compare_with_wikidata#73]
I came across the situation where I added columns to the CSV file a prompt was using (so there were new columns in common with those from the SPARQL) but those new columns didn't appear. This was because the templates generated from the first the prompt ran looked like, for example:

```
<!-- Feel free to edit this template. If you want to get the default back just delete the contents of the page and then refresh the prompt. -->
|-
| {{{@@}}}
| {{{minister}}}
```

It would have been good if the prompter could have detected that this template had not been customized, and so realised it was safe to update it to:

```
<!-- Feel free to edit this template. If you want to get the default back just delete the contents of the page and then refresh the prompt. -->
|-
| {{{@@}}}
| {{{minister}}}
| {{{university}}}
| {{{course}}}
```

### Possible solution: special "safe to rewrite" comment

Perhaps the comment at the top of these templates could be something like:

```
<!-- ALLOW-AUTOMATIC-REPLACEMENT -->
<!-- Feel free to edit this template - if you do, remove the ALLOW-AUTOMATIC-REPLACEMENT
     comment above, otherwise a subsequent run of the tool will overwrite this template.
     If you want to get the default back either put back that comment or delete the
     contents of this page entirely - then in either case refresh the prompt to
     regenerate the default template.
-->
```

### Possible solution: look the revision history for the template

Another alternative (possibly nicer) would be to detect that there had been no user customization by looking at the page history to see if there had been any non-bot changes since the last bot-authored change.

**Comment by mhl** [comment]
Or another option is to just document this in a "Troubleshooting" or "FAQ" document if that's easily findable.

**Comment by lucychambers** [comment]
Decision: documentation

**Comment by chrismytton** [comment]
- [ ] Add FAQ to template page on Wikidata
- [ ] Link to FAQ from Prompter homepage
- [ ] Link to some way to ask questions from prompter (talk page for template?)

**Comment by chrismytton** [comment]
For now I've added an FAQ section to the [template documentation](https://www.wikidata.org/wiki/Template:Compare_Wikidata_with_CSV) along with a link to the talk page where people can ask questions. I've also updated the prompter home page (https://github.com/everypolitician/prompter/commit/4be4e4eda3bccdf3cc1e5462fc6678980538dd5e) to link to that FAQ section.

**Comment by tmtmtmtm** [comment]
I made a few new prompts today, and in each case I ended up needing to add/remove/edit at least one column after the initial version had run. Having to manually blank out four separate templates each time gets very very frustrating very quickly. If we can't have something to auto-detect this case, perhaps we can at least have a 'one click' way to reset them all?

**Comment by chrismytton** [comment]
I've fixed this in #84 by always writing out the default templates, but having them separate from the custom templates. From a user's perspective this should still work in roughly the same way, if you want to go back to the defaults then leave the template blank, only now the check for a blank template is done by Wikidata's templates, rather than Ruby.

This means that if you want to intentionally leave a row blank and not use the defaults, like we often do to hide added/removed rows, then you need to add the following into the appropriate template:

```
<includeonly>__NOINDEX__</includeonly>
```

You can see this in action on [the Heads of Government row added custom template](https://www.wikidata.org/w/index.php?title=User:Chris_Mytton/sandbox/prompts/heads_of_government/row_added_template&action=edit&oldid=549723693).
